src/julia-syntax.scm

```diff
@@ -2939,6 +2953,38 @@
     (lambda (e)
       (set! *current-desugar-loc* e)
       e)
+
+    ;; We insert (latestworld) after every call to _eval_import or _eval_using
+    ;; to avoid having to do it in eval_import_path (#57316)
+    'import
+    (lambda (e)
+      (check-import-paths "import" (cdr e))
+      `(block
+        (toplevel-only import)
+        ,.(if (eq? (caadr e) ':)
+              `((call (core _eval_import) (true) (thismodule)
```

Comment:
nit: `(core)` is usually for builtin-provided (not changeable) things, while `(top)` is usually for our user-definable things (from Base). The added advantage with `(top)` is that different modules could then chose to define a different "top" module (for Core & its submodules it is specified to be Core, while for Base and other modules it defaults to Base)
```suggestion
              `((call (top _eval_import) (true) (thismodule)
```

Reply:
I was wondering what the better option was here. Ultimately I went with core only because we need `_eval_import` inside of `boot.jl` and Base is not defined until later. I'll figure out how to work around that